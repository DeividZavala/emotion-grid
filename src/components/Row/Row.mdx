---
name: Row
---

import { Props } from 'docz';
import { Container } from '../Container';
import { Row } from './';

# Row

<Container debug>
  <Row debug reverse align='center' justify={{ xl: 'flex-end' }} style={{ height: '400px'}}>
    <h1>PROPS</h1>
  </Row>

  <Row debug reverse={['md']} align={{ lg: 'center' }} justify={{ lg: 'center' }} style={{ height: '400px'}}>
    <h1>OBJECT</h1>
  </Row>
</Container>
