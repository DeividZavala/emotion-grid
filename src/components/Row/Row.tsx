import React, { FC } from 'react';

import styled from '@emotion/styled';
import { css } from '@emotion/react';

import { Types } from '../../types';
import { config, constants } from '../../config';
import { media } from '../../utils';

import { Col } from '../Col';

const baseStyle = ({ theme }: Types.StyleProps) => css`
  label: row;

  display: flex;
  flex-wrap: wrap;

  box-sizing: border-box;

  ${constants.BREAKPOINTS.map(
    (breakpoint) => css`
      ${media(breakpoint)} {
        margin-left: -${config(theme).gutter[breakpoint] / 2}rem;
        margin-right: -${config(theme).gutter[breakpoint] / 2}rem;
      }
    `
  )}
`;

const alignStyle = ({ align }: Types.RowProps) =>
  align &&
  (typeof align === 'object'
    ? constants.BREAKPOINTS.map(
        (breakpoint) =>
          align[breakpoint] &&
          css`
            ${media(breakpoint)} {
              align-items: ${align[breakpoint]};
            }
          `
      )
    : css`
        align-items: ${align};
      `);

const justifyStyle = ({ justify }: Types.RowProps) =>
  justify &&
  (typeof justify === 'object'
    ? constants.BREAKPOINTS.map(
        (breakpoint) =>
          justify[breakpoint] &&
          css`
            ${media(breakpoint)} {
              justify-content: ${justify[breakpoint]};
            }
          `
      )
    : css`
        justify-content: ${justify};
      `);

const reverseStyle = ({ reverse }: Types.RowProps) =>
  reverse &&
  (Array.isArray(reverse)
    ? constants.BREAKPOINTS.map(
        (breakpoint) => css`
          ${media(breakpoint)} {
            flex-direction: ${reverse.includes(breakpoint)
              ? 'row-reverse'
              : 'row'};
          }
        `
      )
    : css`
        flex-direction: row-reverse;
      `);

const noGutterStyle = ({ noGutters }: Types.RowProps) =>
  noGutters &&
  css`
    ${Col} {
      padding-left: 0;
      padding-right: 0;
    }
  `;

const BaseRow: FC<Types.RowProps> = ({ style, className, children }) => (
  <div style={style} className={className}>
    {children}
  </div>
);

export const Row = styled(BaseRow)<Types.RowProps>(
  baseStyle,
  alignStyle,
  justifyStyle,
  reverseStyle,
  noGutterStyle
);
