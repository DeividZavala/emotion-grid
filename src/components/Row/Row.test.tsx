import React from 'react';
import { render } from '@testing-library/react';

import { Row } from './Row';
import { Col } from '../Col';

describe('Row', () => {
  const breakpoints = ['xs', 'sm', 'md', 'lg', 'xl'] as const;

  const align = [
    'baseline',
    'center',
    'flex-start',
    'flex-end',
    'stretch',
  ] as const;

  const justify = [
    'center',
    'flex-end',
    'flex-start',
    'space-around',
    'space-between',
  ] as const;

  it('should render with default styles', () => {
    const { container } = render(<Row>Row</Row>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with reverse styles when passed the reverse prop', () => {
    const { container } = render(<Row reverse>Row reverse</Row>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it.each(breakpoints)(
    'should render reverse styles with breakpoint %s',
    (breakpoint) => {
      const { container } = render(
        <Row reverse={[breakpoint]}>Row reverse {breakpoint}</Row>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it.each(align)(
    'should render align styles when passed the align %s prop as string',
    (propertie) => {
      const { container } = render(
        <Row align={propertie}>Row align {propertie}</Row>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it('should render align styles when passed the align prop as object', () => {
    const { container } = render(<Row align={{ md: 'center' }}>Row align</Row>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it.each(justify)(
    'should render justify styles when when passed the justify %s prop as string',
    (propertie) => {
      const { container } = render(
        <Row justify={propertie}>Row justify {propertie}</Row>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it('should render justify styles when passed the align prop as object', () => {
    const { container } = render(
      <Row justify={{ md: 'center' }}>Row justify</Row>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with noGutters styles when passed the noGutters prop', () => {
    const { container } = render(
      <Row noGutters>
        <Col>Col</Col>
      </Row>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
