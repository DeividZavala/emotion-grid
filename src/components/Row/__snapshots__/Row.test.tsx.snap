// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render align styles when passed the align baseline prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row align 
  baseline
</div>
`;

exports[`Row should render align styles when passed the align center prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row align 
  center
</div>
`;

exports[`Row should render align styles when passed the align flex-end prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row align 
  flex-end
</div>
`;

exports[`Row should render align styles when passed the align flex-start prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row align 
  flex-start
</div>
`;

exports[`Row should render align styles when passed the align prop as object 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div
  class="grid-0 grid-1"
>
  Row align
</div>
`;

exports[`Row should render align styles when passed the align stretch prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row align 
  stretch
</div>
`;

exports[`Row should render justify styles when passed the align prop as object 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
  class="grid-0 grid-1"
>
  Row justify
</div>
`;

exports[`Row should render justify styles when when passed the justify center prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row justify 
  center
</div>
`;

exports[`Row should render justify styles when when passed the justify flex-end prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row justify 
  flex-end
</div>
`;

exports[`Row should render justify styles when when passed the justify flex-start prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row justify 
  flex-start
</div>
`;

exports[`Row should render justify styles when when passed the justify space-around prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row justify 
  space-around
</div>
`;

exports[`Row should render justify styles when when passed the justify space-between prop as string 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row justify 
  space-between
</div>
`;

exports[`Row should render reverse styles with breakpoint lg 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:64em) {
  .grid-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="grid-0 grid-1"
>
  Row reverse 
  lg
</div>
`;

exports[`Row should render reverse styles with breakpoint md 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:64em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="grid-0 grid-1"
>
  Row reverse 
  md
</div>
`;

exports[`Row should render reverse styles with breakpoint sm 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:64em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="grid-0 grid-1"
>
  Row reverse 
  sm
</div>
`;

exports[`Row should render reverse styles with breakpoint xl 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:64em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

<div
  class="grid-0 grid-1"
>
  Row reverse 
  xl
</div>
`;

exports[`Row should render reverse styles with breakpoint xs 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:48em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:64em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="grid-0 grid-1"
>
  Row reverse 
  xs
</div>
`;

exports[`Row should render with default styles 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row
</div>
`;

exports[`Row should render with noGutters styles when passed the noGutters prop 1`] = `
.grid-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

.grid-2 .grid-1 {
  padding-left: 0;
  padding-right: 0;
}

.grid-0 {
  position: relative;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100%;
  max-width: 100%;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

<div
  class="grid-2 grid-3"
>
  <div
    class="grid-0 grid-1"
  >
    Col
  </div>
</div>
`;

exports[`Row should render with reverse styles when passed the reverse prop 1`] = `
.grid-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

@media (min-width:1em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
>
  Row reverse
</div>
`;
